src/attribute_type.rs

```diff
@@ -396,567 +307,593 @@ impl SdpAttributeSsrc {
 }
 
 #[derive(Clone)]
-pub enum SdpAttributeValue {
-    Str(String),
-    Int(u64),
-    Vector(Vec<String>),
+pub enum SdpAttribute {
+    BundleOnly,
     Candidate(SdpAttributeCandidate),
+    EndOfCandidates,
     Extmap(SdpAttributeExtmap),
     Fingerprint(SdpAttributeFingerprint),
     Fmtp(SdpAttributeFmtp),
     Group(SdpAttributeGroup),
+    IceLite,
+    IceMismatch,
+    IceOptions(Vec<String>),
+    IcePwd(String),
+    IceUfrag(String),
+    Identity(String),
+    ImageAttr(String),
+    Inactive,
+    Label(String),
+    MaxMessageSize(u64),
+    MaxPtime(u64),
+    Mid(String),
     Msid(SdpAttributeMsid),
+    MsidSemantic(String),
+    Ptime(u64),
+    Rid(String),
+    Recvonly,
     RemoteCandidate(SdpAttributeRemoteCandidate),
     Rtpmap(SdpAttributeRtpmap),
     Rtcp(SdpAttributeRtcp),
     Rtcpfb(SdpAttributeRtcpFb),
+    RtcpMux,
+    RtcpRsize,
     Sctpmap(SdpAttributeSctpmap),
+    SctpPort(u64),
+    Sendonly,
+    Sendrecv,
     Setup(SdpAttributeSetup),
     Simulcast(SdpAttributeSimulcast),
     Ssrc(SdpAttributeSsrc),
+    SsrcGroup(String),
 }
 
-#[derive(Clone)]
-pub struct SdpAttribute {
-    pub name: SdpAttributeType,
-    pub value: Option<SdpAttributeValue>,
+impl FromStr for SdpAttribute {
+    type Err = SdpParserError;
+
+    fn from_str(line: &str) -> Result<SdpAttribute, SdpParserError> {
```

Comment:
If I understand from_str() right it should be `Result<SdpAttribute, Self::Err>`. I guess without it you could declare two different things here.